recipes/boa-ide

```diff
@@ -0,0 +1,4 @@
+(boa-ide
+ :fetcher github
+ :repo "boalang/syntax-highlight"
+ :files ("boa-sc-data.el" "boa-ide.el" "boa-study-config.el"))
```

Comment:
`boa-ide.el` with byte-compile using Emacs 28.1:
```
boa-ide.el:165:52: Warning: Unused lexical variable `symbol-start'
boa-ide.el:166:29: Warning: Unused lexical variable `symbol-end'
In boa-ide-mode:
boa-ide.el:205:14: Warning: malformed interactive specc: (interactive (list
    (if current-prefix-arg (prefix-numeric-value current-prefix-arg) 'toggle))
    list 'boa-mode)
In end of data:
boa-ide.el:212:6: Warning: the function `c-update-modeline' is not known to be
    defined.
boa-ide.el:156:8: Warning: the function `boa-mode' is not known to be defined.
```

`boa-ide.el` with package-lint 20230605.1550:
```
3 issues found:
5:38: error: Package boa-sc-data is not installable.
5:60: error: Package boa-mode is not installable.
6:3: warning: You should include standard keywords: see the variable `finder-known-keywords'.
```
Note this recipe is packaging boa-sc-data.el`

`boa-ide.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- boa-ide.el#L188: There is no need to quote lambdas (neither #' nor ')
- boa-ide.el#L80: It's safer to sharp-quote function names; use `#'`
- boa-ide.el#L83: It's safer to sharp-quote function names; use `#'`
```

`boa-sc-data.el` with checkdoc 0.6.2:
```
boa-sc-data.el:55: First sentence should end with punctuation
boa-sc-data.el:65: First sentence should end with punctuation
boa-sc-data.el:72: First sentence should end with punctuation
boa-sc-data.el:78: First sentence should end with punctuation
boa-sc-data.el:84: First sentence should end with punctuation
```
(Minor, but I think these docstrings are begging to be filled in.)

`boa-sc-data.el` with package-lint 20230605.1550:
```
55 issues found:
5:0: error: Package-Requires outside the main file have no effect.
6:3: warning: You should include standard keywords: see the variable `finder-known-keywords'.
34:0: error: "boa-sc-configuration" doesn't start with package's prefix "boa-ide".
54:0: error: "boa-sc-query" doesn't start with package's prefix "boa-ide".
64:0: error: "boa-sc-processor" doesn't start with package's prefix "boa-ide".
71:0: error: "boa-sc-analysis" doesn't start with package's prefix "boa-ide".
77:0: error: "boa-sc-substitution" doesn't start with package's prefix "boa-ide".
83:0: error: "boa-sc-csv" doesn't start with package's prefix "boa-ide".
94:0: error: "boa-sc-data" doesn't start with package's prefix "boa-ide".
97:0: error: "boa-sc-verbosity" doesn't start with package's prefix "boa-ide".
103:0: error: "boa-sc-maphash" doesn't start with package's prefix "boa-ide".
105:3: error: You should depend on (emacs "25.1") if you need `when-let'.
108:0: error: "boa-sc-convert-analysis" doesn't start with package's prefix "boa-ide".
114:0: error: "boa-sc-convert-substitution" doesn't start with package's prefix "boa-ide".
122:0: error: "boa-sc-convert-csv" doesn't start with package's prefix "boa-ide".
132:0: error: "boa-sc-convert-processor" doesn't start with package's prefix "boa-ide".
139:40: error: You should depend on (emacs "25.1") if you need `when-let'.
143:0: error: "boa-sc-convert-query" doesn't start with package's prefix "boa-ide".
150:34: error: You should depend on (emacs "25.1") if you need `when-let'.
158:0: error: "boa-sc-convert-configuration" doesn't start with package's prefix "boa-ide".
179:0: error: "boa-sc-parse" doesn't start with package's prefix "boa-ide".
184:9: error: You should depend on (emacs "25.1") if you need `save-mark-and-excursion'.
193:28: error: You should depend on (emacs "27.1") if you need `json-parse-buffer'.
210:0: error: "boa-sc-parse-needed-p" doesn't start with package's prefix "boa-ide".
213:10: error: You should depend on (emacs "27.1") if you need `time-convert'.
214:10: error: You should depend on (emacs "27.1") if you need `time-convert'.
214:24: error: You should depend on (emacs "26.1") if you need `file-attribute-modification-time'.
219:0: error: "boa-sc-get-study-config-file" doesn't start with package's prefix "boa-ide".
221:3: error: You should depend on (emacs "28.1") if you need `file-name-concat'.
223:0: error: "boa-sc-get-study-config-buffer" doesn't start with package's prefix "boa-ide".
225:3: error: You should depend on (emacs "25.1") if you need `if-let'.
229:5: error: You should depend on (emacs "25.1") if you need `save-mark-and-excursion'.
233:0: error: "boa-sc-get-data" doesn't start with package's prefix "boa-ide".
239:0: error: "boa-sc-get-project-dir" doesn't start with package's prefix "boa-ide".
248:0: error: "boa-sc-get-or-read-project" doesn't start with package's prefix "boa-ide".
262:0: error: "boa-sc-datasets" doesn't start with package's prefix "boa-ide".
270:0: error: "boa-sc-outputs" doesn't start with package's prefix "boa-ide".
276:45: error: You should depend on (emacs "25.1") if you need `when-let'.
282:0: error: "boa-sc-csvs" doesn't start with package's prefix "boa-ide".
286:25: error: You should depend on (emacs "25.1") if you need `when-let'.
289:45: error: You should depend on (emacs "25.1") if you need `when-let'.
295:0: error: "boa-sc-analyses" doesn't start with package's prefix "boa-ide".
303:0: error: "boa-sc-snippets" doesn't start with package's prefix "boa-ide".
311:0: error: "boa-sc-outputs-query" doesn't start with package's prefix "boa-ide".
318:43: error: You should depend on (emacs "25.1") if you need `when-let'.
324:0: error: "boa-sc-csv-query" doesn't start with package's prefix "boa-ide".
329:25: error: You should depend on (emacs "25.1") if you need `when-let'.
332:43: error: You should depend on (emacs "25.1") if you need `when-let'.
338:0: error: "boa-sc-analyses-query" doesn't start with package's prefix "boa-ide".
344:23: error: You should depend on (emacs "26.1") if you need `when-let*'.
353:0: error: "boa-sc-snippets-query" doesn't start with package's prefix "boa-ide".
355:3: error: You should depend on (emacs "26.1") if you need `when-let*'.
367:0: error: "boa-sc-set-verbose" doesn't start with package's prefix "boa-ide".
386:0: error: "boa-sc-get-verbosity" doesn't start with package's prefix "boa-ide".
403:0: error: "boa-sc-compile" doesn't start with package's prefix "boa-ide".
```

`boa-sc-data.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- boa-sc-data.el#L253: There is no need to quote lambdas (neither #' nor ')
- boa-sc-data.el#L265: There is no need to quote lambdas (neither #' nor ')
- boa-sc-data.el#L273: There is no need to quote lambdas (neither #' nor ')
- boa-sc-data.el#L275: There is no need to quote lambdas (neither #' nor ')
- boa-sc-data.el#L285: There is no need to quote lambdas (neither #' nor ')
- boa-sc-data.el#L288: There is no need to quote lambdas (neither #' nor ')
- boa-sc-data.el#L298: There is no need to quote lambdas (neither #' nor ')
- boa-sc-data.el#L360: There is no need to quote lambdas (neither #' nor ')
```

`boa-study-config.el` with byte-compile using Emacs 28.1:
```
boa-study-config.el:32:1: Error: Cannot open load file: No such file or directory, json-snatcher
```

`boa-study-config.el` with package-lint 20230605.1550:
```
20 issues found:
5:0: error: Package-Requires outside the main file have no effect.
6:3: warning: You should include standard keywords: see the variable `finder-known-keywords'.
40:0: error: "boa-study-config-project-dir" doesn't start with package's prefix "boa-ide".
46:0: error: "boa-study-config--get-path" doesn't start with package's prefix "boa-ide".
54:0: error: "boa-study-config-current-context" doesn't start with package's prefix "boa-ide".
70:3: error: You should depend on (emacs "25.1") if you need `when-let'.
102:0: error: "boa-study-config-completion-at-point" doesn't start with package's prefix "boa-ide".
118:22: error: You should depend on (emacs "25.1") if you need `directory-files-recursively'.
121:22: error: You should depend on (emacs "25.1") if you need `directory-files-recursively'.
127:22: error: You should depend on (emacs "25.1") if you need `directory-files-recursively'.
131:36: error: You should depend on (emacs "25.1") if you need `directory-files-recursively'.
142:0: error: "boa-study-config-runnable-at-point" doesn't start with package's prefix "boa-ide".
159:0: error: "boa-study-config-run" doesn't start with package's prefix "boa-ide".
173:0: error: "boa-study-config-ffap-file" doesn't start with package's prefix "boa-ide".
191:0: error: "boa-study-config-ffap" doesn't start with package's prefix "boa-ide".
201:0: error: "boa-study-config-mode-map" doesn't start with package's prefix "boa-ide".
212:0: error: "boa-study-config-maybe-enable" doesn't start with package's prefix "boa-ide".
222:0: error: "boa-study-config-insinuate" doesn't start with package's prefix "boa-ide".
226:0: error: "boa-study-config-reparse" doesn't start with package's prefix "boa-ide".
230:0: error: "boa-study-config-mode" doesn't start with package's prefix "boa-ide".
```

`boa-study-config.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- boa-study-config.el#L117: There is no need to quote lambdas (neither #' nor ')
- boa-study-config.el#L120: There is no need to quote lambdas (neither #' nor ')
- boa-study-config.el#L126: There is no need to quote lambdas (neither #' nor ')
- boa-study-config.el#L130: There is no need to quote lambdas (neither #' nor ')
- boa-study-config.el#L164: There is no need to quote lambdas (neither #' nor ')
- boa-study-config.el#L165: There is no need to quote lambdas (neither #' nor ')
- boa-study-config.el#L203: There is no need to quote lambdas (neither #' nor ')
- boa-study-config.el#L129: It's safer to sharp-quote function names; use `#'`
- boa-study-config.el#L137: Use `when ...` instead of `unless (null ...)`
```